core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowCompilationResource.scala

```diff
@@ -0,0 +1,67 @@
+package edu.uci.ics.texera.web.resource
+
+import com.typesafe.scalalogging.LazyLogging
+import edu.uci.ics.amber.engine.common.virtualidentity.WorkflowIdentity
+import edu.uci.ics.texera.Utils
+import edu.uci.ics.texera.web.auth.SessionUser
+import edu.uci.ics.texera.web.model.http.response.SchemaPropagationResponse
+import edu.uci.ics.texera.web.model.websocket.request.LogicalPlanPojo
+import edu.uci.ics.texera.workflow.common.WorkflowContext
+import edu.uci.ics.texera.workflow.common.tuple.schema.Attribute
+import edu.uci.ics.texera.workflow.common.workflow.{PhysicalPlan, WorkflowCompiler}
+import io.dropwizard.auth.Auth
+import org.jooq.types.UInteger
+
+import javax.annotation.security.RolesAllowed
+import javax.ws.rs.{Consumes, POST, Path, PathParam, Produces}
+import javax.ws.rs.core.MediaType
+
+case class WorkflowCompilationResponse(
+    physicalPlan: PhysicalPlan,
+    operatorInputSchemas: Map[String, List[Option[List[Attribute]]]],
+    operatorErrors: Map[String, String]
+)
+
+@Consumes(Array(MediaType.APPLICATION_JSON))
+@Produces(Array(MediaType.APPLICATION_JSON))
+@Path("/compilation")
+class WorkflowCompilationResource extends LazyLogging {
+  @POST
+  @Path("/{wid}")
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  def compileWorkflow(
+      workflowStr: String,
+      @PathParam("wid") wid: UInteger,
+      @Auth sessionUser: SessionUser
+  ): WorkflowCompilationResponse = {
+    val logicalPlanPojo = Utils.objectMapper.readValue(workflowStr, classOf[LogicalPlanPojo])
+
+    val context = new WorkflowContext(
+      userId = Option(sessionUser.getUser.getUid),
+      workflowId = WorkflowIdentity(wid.toString.toLong)
+    )
```

Comment:
this is a major comment:

do we still need user id for compilation? previously we require user id for resolving the data file path. I am not sure if the logic is still there or is it replaced with the dataset URI. Ideally compilation should not depend on user information?

Reply:
The `userId` is already removed.